action/action-util/src/main/java/com/powsybl/action/util/LoadScalable.java

```diff
@@ -143,6 +136,31 @@ public double scale(Network n, double asked, ScalingConvention scalingConvention
         LOGGER.info("Change active power setpoint of {} from {} to {} ",
                 l.getId(), oldP0, l.getP0());
 
+        if (constantPowerFactor) {
+            l.setQ0(l.getP0() * oldQ0 / oldP0);
+            LOGGER.info("Change reactive power setpoint of {} from {} to {} ",
+                    l.getId(), oldQ0, l.getQ0());
+        }
+
         return done;
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * If scalingConvention is LOAD, the load active power increases for positive "asked" and decreases inversely
+     * If scalingConvention is GENERATOR, the load active power decreases for positive "asked" and increases inversely
+     */
+    @Override
+    public double scale(Network n, double asked, ScalingConvention scalingConvention) {
+        return scale(n, asked, scalingConvention, false);
+    }
+
+    public double scaleConstantPowerFactor(Network n, double asked, ScalingConvention scalingConvention) {
+        return scale(n, asked, scalingConvention, true);
+    }
+
+    public double scaleConstantPowerFactor(Network n, double asked) {
```

Comment:
Don't forget the `@Override` annotation, this should be in the API and implemented in `ProportionalScalable`. The default implementation should call  `scale`